src/shell/shell_cmds.cpp

```diff
@@ -2242,3 +2243,271 @@ void DOS_Shell::CMD_VOL(char* args)
 	const auto trimmed_label = trim(label);
 	WriteOut(MSG_Get("SHELL_CMD_VOL_OUTPUT"), drive, trimmed_label, high, low);
 }
+
+void DOS_Shell::CMD_MOVE(char* args)
+{
+	HELP("MOVE");
+
+	// TODO: Add support for these flags along with the COPYCMD environment
+	// variable (overwrite prompts) Also currently ignored by CMD_COPY
+	(void)ScanCMDBool(args, "Y");
+	(void)ScanCMDBool(args, "-Y");
+
+	char* rem = ScanCMDRemain(args);
+	if (rem) {
+		WriteOut(MSG_Get("SHELL_ILLEGAL_SWITCH"), rem);
+		return;
+	}
+
+	size_t num_sources = 0;
+	char* raw_sources[32];
+	bool parsing          = true;
+	bool quote            = false;
+	bool stop_next_letter = false;
+	bool comma            = false;
+	char* word            = args;
+
+	// Parse sources (seperated by commas)
+	while (parsing) {
+		if (num_sources >= ARRAY_LEN(raw_sources)) {
+			WriteOut(MSG_Get("SHELL_TOO_MANY_PARAMETERS"));
+			return;
+		}
+		switch (*args) {
+		case '"':
+			quote = !quote;
+			memmove(args, args + 1, strlen(args));
```

Comment:
`memmove` and pointer arithmetics are dangerous, try to rework the code using `std::string` type (BTW, I know there is a lot of legacy code uses these techniques).

Reply:
💯

Reply:
Fixed